### DISCUSSION TITLE
Episode 211: Power, Politics and Misconduct in C++ ✨

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#110] [Episode]
Episode Link: https://adspthepodcast.com/2024/12/06/Episode-211.html

Feel free to write comments or ask questions 🙂

**Comment by kentf** [comment]
This episode was something else, wasn't it? :)

The intro led me to believe it was from a yet another Rust-podcast, but you made it clear from the beginning it was generated content. That transparency is a big plus.

I listened to the whole episode, hoping for some real voices/content. I really considered stopping at the 5 minute mark, but since I was walking, and a glimmer of hope hoped for some "real" content, I let it play.

I'm not sure if it was 3 different summaries stitched together, or if the LLM was going in circles (ha, circle), but within each segment it was going in circles repeating the main points.

It was a bit off-putting listening to some of the pronunciations, _"sea ... plus ... plus"_ and _"'erb sudder"_....

Despite that, as a C++ developer, though I'm not basing my identity on being one, I'm drawn to it. Sure, it has it's flaws, but the language has it's charm. Can things be done better? Absolutely! Do I mind Rust? No. Rust did not rise out of a vacuum, it was created to address a specific problem, memory safety. If C++ can learn from Rust, cool! It's not like Rust and C++ is like the Empire vs New Republic.

C++ and Rust can be friends! Their users too!

As such, unsafe C++ in the technical sense, does not matter in the grand scheme. However, socially unsafe C++ is unacceptable.

Personally, I hope you won't lose the C++-bits of the show. Every developer/programmer makes their own adventure through their languages of interest. Some languages will be of interest, some won't. What matters is the journey and what you are interested in.

PS: No AI/LLMs were disturbed while I wrote this comment :)

**Comment by bruxisma** [comment]
Hello,

This episode appears to be in violation of the copyright license my work is released under. Per the [CC BY-NC-ND 4.0](https://creativecommons.org/licenses/by-nc-nd/4.0/)

> If you remix, transform, or build upon the material, you may not distribute the modified material. 

As you did not distribute the content in an unadapted form, and instead fed it into a transformer (NotebookLM), you are currently violating this deed. Please rectify this, or I will be taking legal measures to protect my copyright.

**Reply by codereport** [reply]
Hey Izzy (@bruxisma), 
Is there a personal email to reach you at? Or some platform I can DM you on?
Conor

**Reply by bruxisma** [reply]
(Conor was able to get a hold of me via email)

**Comment by William-Weston** [comment]
This was the absolute worst episode and I've listen to every one.  A mix of high-school gossip and slander.  Cancel culture and the word police have run amok.  I say this as someone who has supported diversity and the rights of marginalized people for my entire adult life.
I would say more but I'm afraid I might get cancelled.

**Reply by kookjr** [reply]
Agree. I don't know enough about all the facts around this topic, but the tone felt like an influencer lacking real knowledge, one you might hear on TT. IMHO this kind of things adds to the polarization and that seems like the last thing we need from a thoughtful podcast. There was some discussion of this in episode #212 and I was hoping this darker side of AI generated content would be brought up, but no.

**Comment by werelord** [comment]
AI generated content is a hard pass for me..

**Comment by mermerico** [comment]
Hey Conor,

I love ADSP and your code_report content but I didn't think this episode was effective. I listen to the podcast because I enjoy listening to you and your guests / co-hosts and I value your opinion. The NotebookLM summary obviously didn't have you in it. It also had some factual errors / hallucinations, and it was poorly structured and went around in circles. But what I think is worse is that it doesn't tell me anything about what you thought about the original post. You clearly thought it was important, but which parts did you agree with? Which parts are you not sure about? What elements of your personal experience back up or refute points in the post? This episode left me wondering why you chose to make this release. If this post is important enough to talk about for a full episode, then why wasn't important enough for you to take the time to record it? If you wanted to distance yourself from some of its conclusions then why use NotebookLM, a service that creates hosts that only agree with the original content? I really appreciate your quick "get in and get out" style of introductions/outros but I think this episode deserved a bit more context.

Again, I love your work and I'm looking forward to next week's episode!

**Reply by codereport** [reply]
Thanks for you thoughtful response @mermerico. The AI episode was an experiment. If we do something like this again, you will be hearing directing from the hosts.

**Reply by kookjr** [reply]
I found myself frequently thinking "is that true?". It made me realize how much I trust the hosts and how little I trusted this generated content.

What may have been interesting was what prompts and process was used to make the content. Were they able to iterate to get the discussion more what they wanted? Not algorithms related I guess but was hoping this was at the beginning or end of the recording.

**Reply by codereport** [reply]
I just fed the article to NotebookLM, no prompts. It is just a "drag and drop" and click generate.

**Comment by goldfishsound** [comment]
This was painful to listen to.  Stripped of any human emotion and reflection. 
please do not do this again.

**Comment by SKyletoft** [comment]
Ignoring the AI voices (though it's scary how good they've become), this episode was done in extremely poor taste, given the content of the article, even if it gets less serious and a bit unhinged towards the end. Especially so given that Bryce is discussed in it, though just for being a painfully pedantic bureaucrat.

**Comment by rtobar** [comment]
Long-time listener (all episodes!), first time commenter. Thanks a lot for the show! I'm eager every week to see what you'll be discussing and if you'll have any guests.

Like a few other people mentioned, I was also a bit thrown off by the episode format, and would have preferred an in-person discussion about this. The "C. Plus. Plus." pronunciation was indeed weird to hear.

After listening to the first half (I stopped there, I don't think I'll bother with the second) it became clear that the summary wasn't good enough, and that it was missing quite a lot of context, so I jumped in and read the full article (it was long! but worth a read, I found it very insightful, learned more than a few things, and have gathered quite a few more references for further reading, thanks @bruxisma!). In particular, I'm only superficially familiar with the C++ standard from listening to podcasts and following a few blogs, so reading all the ins and outs of some of the persons mentioned there (most of which I didn't know) definitely gives a different lens. I don't think it comes as a surprise though (not a pleasant one in any case), all communities end up having their fair bit of issues. You could turn the old phrase of "there are 2 kinds of programming languages" to "Those with community issues, and those that nobody uses".

I hadn't given much thought to which of the two memory safety proposal has more merit, but after the article I'm more inclined for Sean's -- which seems the more unlikely to go through?

**Comment by JVApen** [comment]
The episode was really different for the podcast. Though I really like the fact that this allowed me to take in the content of the blog post while I bailed out on the textual version of the article.

I wouldn't mind having another episode like this which comes out of a (couple of) article(s).

**Comment by robbiesri** [comment]
Interesting to go thru this experience. I think the blog post and reddit thread were certainly...hard to grok, esp with most people's busy lives. So having a summary was a good way to test the waters for this information.

Out of curiosity, did either Conor or Bryce read the entirety of both sources? Or just skimmed + use NotebookLM?

**Reply by codereport** [reply]
I can't speak for Bryce but I read the whole thing.

**Comment by meisam** [comment]
Hello ADSP,

Thanks for the great podcast.

Please invite Andreas Stefik as a guest to discuss designing ergonomic programming languages **based on data**.

Don't ask me how I got to Stefik's work from this episode. It's been a massive rabbit whole that started a year ago when I listened to this episode.

Meisam 🌲
